jerry-ext/debugger/debugger-bluetooth.c

```diff
@@ -0,0 +1,265 @@
+} /* jerryx_debugger_bluetooth_log_error */
+
+/**
+ * Close a bluetooth connection.
+ */
+static void
+jerryx_debugger_bluetooth_close (jerry_debugger_transport_header_t *header_p) /**< bluetooth implementation */
+{
+  JERRYX_ASSERT (!jerry_debugger_transport_is_connected ());
+
+  jerryx_debugger_transport_bluetooth_t *bluetooth_p = (jerryx_debugger_transport_bluetooth_t *) header_p;
+
+  JERRYX_DEBUG_MSG ("bluetooth connection closed\n");
+
+  close (bluetooth_p->bluetooth_socket);
+
+  jerry_heap_free ((void *) header_p, sizeof (jerryx_debugger_transport_bluetooth_t));
+} /* jerryx_debugger_bluetooth_close */
+
+/**
+ * Send data over a bluetooth connection.
+ *
+ * @return true - if the data has been sent successfully
+ *         false - otherwise
+ */
+static bool
+jerryx_debugger_bluetooth_send (jerry_debugger_transport_header_t *header_p, /**< bluetooth implementation */
+                          uint8_t *message_p, /**< message to be sent */
+                          size_t message_length) /**< message length in bytes */
+{
+  JERRYX_ASSERT (jerry_debugger_transport_is_connected ());
+
+  jerryx_debugger_transport_bluetooth_t *bluetooth_p = (jerryx_debugger_transport_bluetooth_t *) header_p;
+
+  do
+  {
+    ssize_t is_err = recv (bluetooth_p->bluetooth_socket, NULL, 0, MSG_PEEK);
+
+    if (is_err == 0 && errno != EWOULDBLOCK)
+    {
+      int err_val = errno;
+      jerry_debugger_transport_close ();
+      jerryx_debugger_bluetooth_log_error (err_val);
+      return false;
+    }
+    ssize_t sent_bytes = send (bluetooth_p->bluetooth_socket, message_p, message_length, 0);
+
+    if (sent_bytes < 0)
+    {
+      if (errno == EWOULDBLOCK)
+      {
+        continue;
+      }
+
+      int err_val = errno;
+      jerry_debugger_transport_close ();
+      jerryx_debugger_bluetooth_log_error (err_val);
+      return false;
+    }
+
+    message_p += sent_bytes;
+    message_length -= (size_t) sent_bytes;
+  }
+  while (message_length > 0);
+
+  return true;
+} /* jerryx_debugger_bluetooth_send */
+
+/**
+ * Receive data from a bluetooth connection.
+ */
+static bool
+jerryx_debugger_bluetooth_receive (jerry_debugger_transport_header_t *header_p, /**< bluetooth implementation */
+                             jerry_debugger_transport_receive_context_t *receive_context_p) /**< receive context */
+{
+  jerryx_debugger_transport_bluetooth_t *bluetooth_p = (jerryx_debugger_transport_bluetooth_t *) header_p;
+
+  uint8_t *buffer_p = receive_context_p->buffer_p + receive_context_p->received_length;
+  size_t buffer_size = JERRY_DEBUGGER_TRANSPORT_MAX_BUFFER_SIZE - receive_context_p->received_length;
+
+  ssize_t length = recv (bluetooth_p->bluetooth_socket, buffer_p, buffer_size, 0);
+
+  if (length <= 0)
+  {
+    if (errno != EWOULDBLOCK || length == 0)
+    {
+      int err_val = errno;
+      jerry_debugger_transport_close ();
+      jerryx_debugger_bluetooth_log_error (err_val);
```

Comment:
Each `jerry_debugger_transport_close ` is followed by `jerryx_debugger_bluetooth_log_error (err_val)`. How about adding the logging function to the end of the transport close function? Therefore it would be less function calls.